In [1]:
import numpy as np 
import pandas as pd 

In [2]:
import json 
from sklearn import preprocessing

In [3]:
import matplotlib.pyplot as plt

In [4]:
import math as m 

In [5]:
%load_ext ipython_memory_usage

Enabling IPython Memory Usage, use %imu_start to begin, %imu_stop to end


In [6]:
%imu_start 

'IPython Memory Usage started'

In [6] used 0.4 MiB RAM in 0.12s (system mean cpu 0%, single max cpu 0%), peaked 0.0 MiB above final usage, current RAM usage now 154.5 MiB


### EDA de la dataFrame 'Athlete_events'

In [7]:
df_athlete=pd.read_csv('data/athlete_events.csv',sep=',')

df_athlete

Name Sex   Age  Height      Team  NOC        Games  \
0          AALERUD Katrine   F  27.0     NaN    Norway  NOR   Tokyo 2020   
1              ABAD Nestor   M  28.0   165.0     Spain  ESP   Tokyo 2020   
2        ABAGNALE Giovanni   M  26.0   198.0     Italy  ITA   Tokyo 2020   
3           ABALDE Alberto   M  26.0   200.0     Spain  ESP   Tokyo 2020   
4            ABALDE Tamara   F  32.0   192.0     Spain  ESP   Tokyo 2020   
...                    ...  ..   ...     ...       ...  ...          ...   
282985          Andrzej ya   M  29.0   179.0  Poland-1  POL  1976 Winter   
282986            Piotr ya   M  27.0   176.0    Poland  POL  2014 Winter   
282987            Piotr ya   M  27.0   176.0    Poland  POL  2014 Winter   
282988  Tomasz Ireneusz ya   M  30.0   185.0    Poland  POL  1998 Winter   
282989  Tomasz Ireneusz ya   M  34.0   185.0    Poland  POL  2002 Winter   

        Year  Season            City                Sport Medal  \
0       2020  Summer           Tokyo         Cycling Road   NaN   
1       2020  Summer           Tokyo  Artistic Gymnastics   NaN   
2       2020  Summer           Tokyo               Rowing   NaN   
3       2020  Summer           Tokyo           Basketball   NaN   
4       2020  Summer           Tokyo           Basketball   NaN   
...      ...     ...             ...                  ...   ...   
282985  1976  Winter       Innsbruck                 Luge   NaN   
282986  2014  Winter           Sochi          Ski Jumping   NaN   
282987  2014  Winter           Sochi          Ski Jumping   NaN   
282988  1998  Winter          Nagano            Bobsleigh   NaN   
282989  2002  Winter  Salt Lake City            Bobsleigh   NaN   

                                           Event        ID  Weight  \
0                                            NaN       NaN     NaN   
1                                            NaN       NaN     NaN   
2                                            NaN       NaN     NaN   
3                                            NaN       NaN     NaN   
4                                            NaN       NaN     NaN   
...                                          ...       ...     ...   
282985                Luge Mixed (Men)'s Doubles  135569.0    89.0   
282986  Ski Jumping Men's Large Hill, Individual  135570.0    59.0   
282987        Ski Jumping Men's Large Hill, Team  135570.0    59.0   
282988                      Bobsleigh Men's Four  135571.0    96.0   
282989                      Bobsleigh Men's Four  135571.0    96.0   

                     Country_org  
0                          Japan  
1                          Japan  
2                          Japan  
3                          Japan  
4                          Japan  
...                          ...  
282985                    Canada  
282986        Russian Federation  
282987        Russian Federation  
282988                     Japan  
282989  United States of America  

[282990 rows x 16 columns]

In [7] used 103.0 MiB RAM in 7.37s (system mean cpu 38%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 257.5 MiB


### DataFrame dictionnary

In [8]:
df_dictionnary=pd.read_csv('data/dictionary.csv')

df_dictionnary=df_dictionnary.rename(columns={'GDP per Capita':'PIB/habitant','Code':'NOC'})

df_dictionnary.isna().sum()

Country          0
NOC              0
Population       5
PIB/habitant    25
dtype: int64

In [8] used -43.5 MiB RAM in 3.05s (system mean cpu 34%, single max cpu 100%), peaked 43.4 MiB above final usage, current RAM usage now 214.0 MiB


### On cherche les athlètes qui n'ont pas de team et ceux des îles TUVALU

In [9]:
df_athlete[df_athlete['NOC']=='UNK']


Name Sex  Age  Height     Team  NOC        Games  Year  \
72954   Fritz Eccard   M  NaN     NaN  Unknown  UNK  1912 Summer  1912   
142595     A. Laffen   M  NaN     NaN  Unknown  UNK  1912 Summer  1912   

        Season       City             Sport Medal  \
72954   Summer  Stockholm  Art Competitions   NaN   
142595  Summer  Stockholm  Art Competitions   NaN   

                                      Event       ID  Weight Country_org  
72954   Art Competitions Mixed Architecture  31292.0     NaN      Sweden  
142595  Art Competitions Mixed Architecture  65813.0     NaN      Sweden

In [9] used 3.3 MiB RAM in 0.15s (system mean cpu 35%, single max cpu 83%), peaked 0.0 MiB above final usage, current RAM usage now 217.3 MiB


Fritez Eccard a participé dans l'équipe Suédoise et A.Laffen .... aucunes traces je sais pas qui est ce mec 

In [10]:
df_athlete[df_athlete['NOC']=='ROT'].nunique()


Name           10
Sex             2
Age             7
Height          7
Team            1
NOC             1
Games           1
Year            1
Season          1
City            1
Sport           3
Medal           0
Event          12
ID             10
Weight          9
Country_org     1
dtype: int64

In [10] used 1.0 MiB RAM in 0.14s (system mean cpu 28%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 218.3 MiB


12 participations parmi 10 athlètes qui ont participé aux JO ces dernières années sous la team des athlètes réfugiés. 

In [11]:
df_athlete=df_athlete.query(" NOC !='ROC' and NOC != 'UNK' ")

In [11] used 40.9 MiB RAM in 0.18s (system mean cpu 40%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 259.2 MiB


## Base données région 

In [12]:
df_regions=pd.read_csv('data/noc_regions.csv')

df_regions=df_regions.rename(columns={'region':'Country'})
df_regions

NOC      Country                 notes
0    AFG  Afghanistan                   NaN
1    AHO      Curacao  Netherlands Antilles
2    ALB      Albania                   NaN
3    ALG      Algeria                   NaN
4    AND      Andorra                   NaN
..   ...          ...                   ...
225  YEM        Yemen                   NaN
226  YMD        Yemen           South Yemen
227  YUG       Serbia            Yugoslavia
228  ZAM       Zambia                   NaN
229  ZIM     Zimbabwe                   NaN

[230 rows x 3 columns]

In [12] used -1.0 MiB RAM in 0.65s (system mean cpu 29%, single max cpu 100%), peaked 0.2 MiB above final usage, current RAM usage now 258.1 MiB


In [13]:
df_regions.nunique()

NOC        230
Country    206
notes       21
dtype: int64

In [13] used 0.0 MiB RAM in 0.11s (system mean cpu 35%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 258.1 MiB


Il y a des régions qui ont le même NOC 

In [14]:
duplicates=[df_regions.duplicated(subset='Country')]

In [14] used 0.0 MiB RAM in 0.11s (system mean cpu 21%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 258.2 MiB


In [15]:
df_regions[df_regions.duplicated(subset='Country')]

NOC         Country                   notes
12   AUS       Australia                     NaN
54   CZE  Czech Republic                     NaN
76   GDR         Germany                     NaN
79   GER         Germany                     NaN
81   GRE          Greece                     NaN
88   HKG           China               Hong Kong
126  MAS        Malaysia                     NaN
143  NBO        Malaysia            North Borneo
147  NFL          Canada            Newfoundland
171  RUS          Russia                     NaN
173  SAA         Germany                     NaN
185  SRB          Serbia                     NaN
197  TCH  Czech Republic                     NaN
208  TUV             NaN                  Tuvalu
210  UAR           Syria    United Arab Republic
213  UNK             NaN                 Unknown
214  URS          Russia                     NaN
222  VNM         Vietnam                     NaN
223  WIF        Trinidad  West Indies Federation
225  YEM           Yemen                     NaN
226  YMD           Yemen             South Yemen
227  YUG          Serbia              Yugoslavia
229  ZIM        Zimbabwe                     NaN

In [15] used 0.1 MiB RAM in 0.12s (system mean cpu 30%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 258.2 MiB


In [16]:
df_regions.isnull().sum()

NOC          0
Country      3
notes      209
dtype: int64

In [16] used -1.3 MiB RAM in 0.11s (system mean cpu 27%, single max cpu 100%), peaked 1.3 MiB above final usage, current RAM usage now 256.9 MiB


In [17]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [17] used -6.8 MiB RAM in 0.11s (system mean cpu 12%, single max cpu 100%), peaked 6.8 MiB above final usage, current RAM usage now 250.2 MiB


Il y a  3 des regions qui ne sont pas attribuées, ROT ( refugee olympic team)  , TUVALU ( 9 archipels qui constituent un État ) et UNK qui correspond à Unknown 

### On ajoute les iles TUVALU 

In [18]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [18] used 0.1 MiB RAM in 0.12s (system mean cpu 25%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 250.2 MiB


In [19]:
df_regions=df_regions.query(" NOC != 'ROT' and NOC != 'UNK' ")

df_regions.isnull().sum()

NOC          0
Country      1
notes      209
dtype: int64

In [19] used 0.0 MiB RAM in 0.12s (system mean cpu 17%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 250.3 MiB


In [20]:
df_regions['Country']=df_regions['Country'].fillna('Tuvalu')

In [20] used 0.3 MiB RAM in 0.10s (system mean cpu 20%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 250.6 MiB


<ipython-input-20-0f7f308fe64b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_regions['Country']=df_regions['Country'].fillna('Tuvalu')


In [21]:
df_regions[df_regions['Country']=='Tuvalu']

NOC Country   notes
208  TUV  Tuvalu  Tuvalu

In [21] used 0.0 MiB RAM in 0.11s (system mean cpu 31%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 250.6 MiB


In [22]:
df_regions.isnull().sum()

NOC          0
Country      0
notes      209
dtype: int64

In [22] used -1.1 MiB RAM in 0.11s (system mean cpu 21%, single max cpu 100%), peaked 1.1 MiB above final usage, current RAM usage now 249.5 MiB


# On va merge les dataframes 

## Merge en left 

In [23]:
df_complete=pd.merge(pd.merge(df_athlete,df_regions,on='NOC',how='left'),df_dictionnary,on='NOC',how='left')
#df_complete=df_complete.drop(labels='',axis=1)

df_complete

Name Sex   Age  Height      Team  NOC        Games  \
0          AALERUD Katrine   F  27.0     NaN    Norway  NOR   Tokyo 2020   
1              ABAD Nestor   M  28.0   165.0     Spain  ESP   Tokyo 2020   
2        ABAGNALE Giovanni   M  26.0   198.0     Italy  ITA   Tokyo 2020   
3           ABALDE Alberto   M  26.0   200.0     Spain  ESP   Tokyo 2020   
4            ABALDE Tamara   F  32.0   192.0     Spain  ESP   Tokyo 2020   
...                    ...  ..   ...     ...       ...  ...          ...   
282622          Andrzej ya   M  29.0   179.0  Poland-1  POL  1976 Winter   
282623            Piotr ya   M  27.0   176.0    Poland  POL  2014 Winter   
282624            Piotr ya   M  27.0   176.0    Poland  POL  2014 Winter   
282625  Tomasz Ireneusz ya   M  30.0   185.0    Poland  POL  1998 Winter   
282626  Tomasz Ireneusz ya   M  34.0   185.0    Poland  POL  2002 Winter   

        Year  Season            City  ... Medal  \
0       2020  Summer           Tokyo  ...   NaN   
1       2020  Summer           Tokyo  ...   NaN   
2       2020  Summer           Tokyo  ...   NaN   
3       2020  Summer           Tokyo  ...   NaN   
4       2020  Summer           Tokyo  ...   NaN   
...      ...     ...             ...  ...   ...   
282622  1976  Winter       Innsbruck  ...   NaN   
282623  2014  Winter           Sochi  ...   NaN   
282624  2014  Winter           Sochi  ...   NaN   
282625  1998  Winter          Nagano  ...   NaN   
282626  2002  Winter  Salt Lake City  ...   NaN   

                                           Event        ID  Weight  \
0                                            NaN       NaN     NaN   
1                                            NaN       NaN     NaN   
2                                            NaN       NaN     NaN   
3                                            NaN       NaN     NaN   
4                                            NaN       NaN     NaN   
...                                          ...       ...     ...   
282622                Luge Mixed (Men)'s Doubles  135569.0    89.0   
282623  Ski Jumping Men's Large Hill, Individual  135570.0    59.0   
282624        Ski Jumping Men's Large Hill, Team  135570.0    59.0   
282625                      Bobsleigh Men's Four  135571.0    96.0   
282626                      Bobsleigh Men's Four  135571.0    96.0   

                     Country_org Country_x notes Country_y  Population  \
0                          Japan    Norway   NaN    Norway   5195921.0   
1                          Japan     Spain   NaN     Spain  46418269.0   
2                          Japan     Italy   NaN     Italy  60802085.0   
3                          Japan     Spain   NaN     Spain  46418269.0   
4                          Japan     Spain   NaN     Spain  46418269.0   
...                          ...       ...   ...       ...         ...   
282622                    Canada    Poland   NaN    Poland  37999494.0   
282623        Russian Federation    Poland   NaN    Poland  37999494.0   
282624        Russian Federation    Poland   NaN    Poland  37999494.0   
282625                     Japan    Poland   NaN    Poland  37999494.0   
282626  United States of America    Poland   NaN    Poland  37999494.0   

        PIB/habitant  
0       74400.369777  
1       25831.582305  
2       29957.804315  
3       25831.582305  
4       25831.582305  
...              ...  
282622  12554.547554  
282623  12554.547554  
282624  12554.547554  
282625  12554.547554  
282626  12554.547554  

[282627 rows x 21 columns]

In [23] used 109.5 MiB RAM in 0.36s (system mean cpu 35%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 359.0 MiB


In [24]:
df_complete=df_complete.query(" NOC !='ROC' ")

In [24] used -8.8 MiB RAM in 0.15s (system mean cpu 42%, single max cpu 100%), peaked 8.8 MiB above final usage, current RAM usage now 350.2 MiB


In [25]:
#df_complete.info()

In [25] used -10.6 MiB RAM in 0.11s (system mean cpu 56%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 339.6 MiB


282 627 lignes , c'est normal on a enlevé les  réfugiés

In [26]:
df_complete.loc[(df_complete['Medal'].isna(),'Medal')]='Pas_medaille'



In [26] used 0.3 MiB RAM in 0.13s (system mean cpu 45%, single max cpu 100%), peaked 0.5 MiB above final usage, current RAM usage now 339.8 MiB


## Encoding test 

In [27]:
dictio_score_medaille={'Pas_medaille':0,
                      'Bronze':1,
                      'Silver':2,
                      'Gold':3}

df_complete['Score_medaille']=df_complete['Medal'].map(dictio_score_medaille)

df_complete

Name Sex   Age  Height      Team  NOC        Games  \
0          AALERUD Katrine   F  27.0     NaN    Norway  NOR   Tokyo 2020   
1              ABAD Nestor   M  28.0   165.0     Spain  ESP   Tokyo 2020   
2        ABAGNALE Giovanni   M  26.0   198.0     Italy  ITA   Tokyo 2020   
3           ABALDE Alberto   M  26.0   200.0     Spain  ESP   Tokyo 2020   
4            ABALDE Tamara   F  32.0   192.0     Spain  ESP   Tokyo 2020   
...                    ...  ..   ...     ...       ...  ...          ...   
282622          Andrzej ya   M  29.0   179.0  Poland-1  POL  1976 Winter   
282623            Piotr ya   M  27.0   176.0    Poland  POL  2014 Winter   
282624            Piotr ya   M  27.0   176.0    Poland  POL  2014 Winter   
282625  Tomasz Ireneusz ya   M  30.0   185.0    Poland  POL  1998 Winter   
282626  Tomasz Ireneusz ya   M  34.0   185.0    Poland  POL  2002 Winter   

        Year  Season            City  ...  \
0       2020  Summer           Tokyo  ...   
1       2020  Summer           Tokyo  ...   
2       2020  Summer           Tokyo  ...   
3       2020  Summer           Tokyo  ...   
4       2020  Summer           Tokyo  ...   
...      ...     ...             ...  ...   
282622  1976  Winter       Innsbruck  ...   
282623  2014  Winter           Sochi  ...   
282624  2014  Winter           Sochi  ...   
282625  1998  Winter          Nagano  ...   
282626  2002  Winter  Salt Lake City  ...   

                                           Event        ID Weight  \
0                                            NaN       NaN    NaN   
1                                            NaN       NaN    NaN   
2                                            NaN       NaN    NaN   
3                                            NaN       NaN    NaN   
4                                            NaN       NaN    NaN   
...                                          ...       ...    ...   
282622                Luge Mixed (Men)'s Doubles  135569.0   89.0   
282623  Ski Jumping Men's Large Hill, Individual  135570.0   59.0   
282624        Ski Jumping Men's Large Hill, Team  135570.0   59.0   
282625                      Bobsleigh Men's Four  135571.0   96.0   
282626                      Bobsleigh Men's Four  135571.0   96.0   

                     Country_org  Country_x notes Country_y  Population  \
0                          Japan     Norway   NaN    Norway   5195921.0   
1                          Japan      Spain   NaN     Spain  46418269.0   
2                          Japan      Italy   NaN     Italy  60802085.0   
3                          Japan      Spain   NaN     Spain  46418269.0   
4                          Japan      Spain   NaN     Spain  46418269.0   
...                          ...        ...   ...       ...         ...   
282622                    Canada     Poland   NaN    Poland  37999494.0   
282623        Russian Federation     Poland   NaN    Poland  37999494.0   
282624        Russian Federation     Poland   NaN    Poland  37999494.0   
282625                     Japan     Poland   NaN    Poland  37999494.0   
282626  United States of America     Poland   NaN    Poland  37999494.0   

        PIB/habitant  Score_medaille  
0       74400.369777               0  
1       25831.582305               0  
2       29957.804315               0  
3       25831.582305               0  
4       25831.582305               0  
...              ...             ...  
282622  12554.547554               0  
282623  12554.547554               0  
282624  12554.547554               0  
282625  12554.547554               0  
282626  12554.547554               0  

[282627 rows x 22 columns]

In [27] used 10.9 MiB RAM in 0.29s (system mean cpu 48%, single max cpu 100%), peaked 0.8 MiB above final usage, current RAM usage now 350.8 MiB


In [28]:
#Marche aussi 
''' 
medaille_categories=['Pas_medaille','Bronze','Silver','Gold']
encode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])

df_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])

df_complete
''' 

" \nmedaille_categories=['Pas_medaille','Bronze','Silver','Gold']\nencode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])\n\ndf_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])\n\ndf_complete\n"

In [28] used 0.1 MiB RAM in 0.11s (system mean cpu 78%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 350.8 MiB


## Analyse valeurs nulles 

In [29]:
df_complete.isna().sum()

Name                   0
Sex                  156
Age                 9628
Height             67142
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Medal                  0
Event               9278
ID                 11513
Weight             74386
Country_org         1733
Country_x            419
notes             277510
Country_y          26221
Population         27593
PIB/habitant       35919
Score_medaille         0
dtype: int64

In [29] used -36.1 MiB RAM in 0.25s (system mean cpu 35%, single max cpu 100%), peaked 36.1 MiB above final usage, current RAM usage now 314.7 MiB


156 sports où on sait pas ce que c'est .... et des country (26 221) qui ne sont pas attribués non plus 

Pour les Country , on va merge 

In [30]:
#df_complete_test=df_complete.copy()

df_complete=pd.merge(df_complete,df_regions,on='NOC',how='left')

df_complete['Country']=df_complete['Country_y'].fillna(df_complete['Country_x'])

In [30] used 25.6 MiB RAM in 0.18s (system mean cpu 23%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 340.3 MiB


In [31]:
df_complete= df_complete.drop(['Country_x', 'Country_y'], axis=1)

In [31] used 18.2 MiB RAM in 0.16s (system mean cpu 56%, single max cpu 100%), peaked 2.9 MiB above final usage, current RAM usage now 358.5 MiB


In [32]:
liste_pays=df_regions['Country'].tolist()
liste_NOC=df_regions['NOC'].tolist()

dictio_NOC_pays=dict(zip(liste_NOC,liste_pays))

dictio_NOC_pays['SGP']='Singapour'
dictio_NOC_pays['LBN']='Liban'


df_complete['Country']=df_complete['Country'].fillna(df_complete['NOC'].map(dictio_NOC_pays))
#liste_region

#df_complete_test.isna().sum()

In [32] used -1.2 MiB RAM in 0.14s (system mean cpu 62%, single max cpu 100%), peaked 1.2 MiB above final usage, current RAM usage now 357.3 MiB


In [33]:
#test=df_complete_test[df_complete_test['Country'].isna()]

#test['Country']=test['Country'].fillna(test['NOC'].map(dictio_NOC_pays))

#test.groupby('NOC')['Name'].count().reset_index()

In [33] used -32.0 MiB RAM in 0.10s (system mean cpu 51%, single max cpu 100%), peaked 30.7 MiB above final usage, current RAM usage now 325.3 MiB


Il y a quatre NOC qui ne correspondent à aucun pays dans notre EOR : réfugiés , ROT réfugiés aussi , LBN c'est le Liban (Lebanon en anglais )

In [34]:
df_complete=df_complete.query(" NOC !='ROT' and NOC !='EOR' ")



In [34] used -42.4 MiB RAM in 0.20s (system mean cpu 48%, single max cpu 100%), peaked 35.1 MiB above final usage, current RAM usage now 282.9 MiB


In [35]:
df_complete=df_complete.drop(['notes_x','notes_y'],axis=1)

In [35] used -34.1 MiB RAM in 0.16s (system mean cpu 30%, single max cpu 100%), peaked 34.1 MiB above final usage, current RAM usage now 248.8 MiB


## Opérations sur les sports 

In [36]:
liste_sports=df_complete['Sport'].unique().tolist()

print(liste_sports)

['Cycling Road', 'Artistic Gymnastics', 'Rowing', 'Basketball', 'Swimming', 'Karate', 'Wrestling', 'Handball', 'Rhythmic Gymnastics', 'Athletics', 'Judo', 'Shooting', 'Table Tennis', 'Football', 'Taekwondo', 'Fencing', 'Badminton', 'Boxing', 'Archery', 'Beach Volleyball', 'Sailing', 'Weightlifting', 'Hockey', 'Trampoline Gymnastics', 'Triathlon', 'Canoe Slalom', 'Water Polo', 'Surfing', 'Canoe Sprint', 'Rugby Sevens', 'Volleyball', nan, 'Equestrian', 'Artistic Swimming', 'Cycling Track', 'Golf', 'Skateboarding', 'Marathon Swimming', 'Cycling BMX Racing', 'Tennis', 'Baseball/Softball', 'Diving', 'Cycling Mountain Bike', '3x3 Basketball', 'Modern Pentathlon', 'Cycling BMX Freestyle', 'Sport Climbing', 'Tug-Of-War', 'Speed Skating', 'Cross Country Skiing', 'Ice Hockey', 'Biathlon', 'Gymnastics', 'Art Competitions', 'Alpine Skiing', 'Luge', 'Bobsleigh', 'Equestrianism', 'Cycling', 'Canoeing', 'Figure Skating', 'Softball', 'Synchronized Swimming', 'Nordic Combined', 'Baseball', 'Freestyle S

In [37]:
table_sport=df_complete.groupby('Sport')['Name'].count().reset_index()

table_sport

Sport  Name
0   3x3 Basketball    56
1      Aeronautics     1
2    Alpine Skiing  8829
3         Alpinism    25
4          Archery  2461
..             ...   ...
79      Tug-Of-War   170
80      Volleyball  3666
81      Water Polo  4119
82   Weightlifting  4128
83       Wrestling  7425

[84 rows x 2 columns]

In [37] used -21.7 MiB RAM in 0.16s (system mean cpu 41%, single max cpu 100%), peaked 21.7 MiB above final usage, current RAM usage now 227.3 MiB


In [38]:
df_complete=df_complete[df_complete['Year']>=2000]

In [38] used 2.2 MiB RAM in 0.13s (system mean cpu 13%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 229.5 MiB


In [39]:
df_complete

Name Sex   Age  Height     Team  NOC  \
0                    AALERUD Katrine   F  27.0     NaN   Norway  NOR   
1                        ABAD Nestor   M  28.0   165.0    Spain  ESP   
2                  ABAGNALE Giovanni   M  26.0   198.0    Italy  ITA   
3                     ABALDE Alberto   M  26.0   200.0    Spain  ESP   
4                      ABALDE Tamara   F  32.0   192.0    Spain  ESP   
...                              ...  ..   ...     ...      ...  ...   
282620  Aleksandr Viktorovich Zyuzin   M  28.0   183.0   Russia  RUS   
282621        Olga Igorevna Zyuzkova   F  33.0   171.0  Belarus  BLR   
282623                      Piotr ya   M  27.0   176.0   Poland  POL   
282624                      Piotr ya   M  27.0   176.0   Poland  POL   
282626            Tomasz Ireneusz ya   M  34.0   185.0   Poland  POL   

              Games  Year  Season            City                Sport  \
0        Tokyo 2020  2020  Summer           Tokyo         Cycling Road   
1        Tokyo 2020  2020  Summer           Tokyo  Artistic Gymnastics   
2        Tokyo 2020  2020  Summer           Tokyo               Rowing   
3        Tokyo 2020  2020  Summer           Tokyo           Basketball   
4        Tokyo 2020  2020  Summer           Tokyo           Basketball   
...             ...   ...     ...             ...                  ...   
282620  2004 Summer  2004  Summer          Athina               Rowing   
282621  2016 Summer  2016  Summer  Rio de Janeiro           Basketball   
282623  2014 Winter  2014  Winter           Sochi          Ski Jumping   
282624  2014 Winter  2014  Winter           Sochi          Ski Jumping   
282626  2002 Winter  2002  Winter  Salt Lake City            Bobsleigh   

               Medal                                     Event        ID  \
0       Pas_medaille                                       NaN       NaN   
1       Pas_medaille                                       NaN       NaN   
2       Pas_medaille                                       NaN       NaN   
3       Pas_medaille                                       NaN       NaN   
4       Pas_medaille                                       NaN       NaN   
...              ...                                       ...       ...   
282620  Pas_medaille    Rowing Men's Lightweight Coxless Fours  135567.0   
282621  Pas_medaille             Basketball Women's Basketball  135568.0   
282623  Pas_medaille  Ski Jumping Men's Large Hill, Individual  135570.0   
282624  Pas_medaille        Ski Jumping Men's Large Hill, Team  135570.0   
282626  Pas_medaille                      Bobsleigh Men's Four  135571.0   

        Weight               Country_org   Population  PIB/habitant  \
0          NaN                     Japan    5195921.0  74400.369777   
1          NaN                     Japan   46418269.0  25831.582305   
2          NaN                     Japan   60802085.0  29957.804315   
3          NaN                     Japan   46418269.0  25831.582305   
4          NaN                     Japan   46418269.0  25831.582305   
...        ...                       ...          ...           ...   
282620    72.0                    Greece  144096812.0   9092.580536   
282621    69.0                    Brazil    9513000.0   5740.456495   
282623    59.0        Russian Federation   37999494.0  12554.547554   
282624    59.0        Russian Federation   37999494.0  12554.547554   
282626    96.0  United States of America   37999494.0  12554.547554   

        Score_medaille  Country  
0                    0   Norway  
1                    0    Spain  
2                    0    Italy  
3                    0    Spain  
4                    0    Spain  
...                ...      ...  
282620               0   Russia  
282621               0  Belarus  
282623               0   Poland  
282624               0   Poland  
282626               0   Poland  

[96730 rows x 20 columns]

In [39] used 0.2 MiB RAM in 0.13s (system mean cpu 25%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 229.7 MiB


## Opérations sur les Ages

In [40]:
table_sport_age=df_complete.groupby('Sport')['Age'].mean().apply(m.ceil).reset_index()

table_sport_age

Sport  Age
0        3x3 Basketball   30
1         Alpine Skiing   25
2               Archery   27
3   Artistic Gymnastics   25
4     Artistic Swimming   24
..                  ...  ...
64            Triathlon   29
65           Volleyball   27
66           Water Polo   27
67        Weightlifting   25
68            Wrestling   27

[69 rows x 2 columns]

In [40] used -2.5 MiB RAM in 0.13s (system mean cpu 35%, single max cpu 100%), peaked 0.7 MiB above final usage, current RAM usage now 227.2 MiB


In [41]:
liste_sport=table_sport_age['Sport'].tolist()
liste_age=table_sport_age['Age'].tolist()

dictio_age_sport=dict(zip(liste_sport,liste_age))

In [41] used 0.1 MiB RAM in 0.11s (system mean cpu 39%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 227.2 MiB


In [42]:
df_complete['Age']=df_complete['Age'].fillna(df_complete['Sport'].map(dictio_age_sport))

df_complete.isna().sum()

Name                  0
Sex                 156
Age                 156
Height             7658
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport               156
Medal                 0
Event              9249
ID                11484
Weight            12694
Country_org           0
Population         2855
PIB/habitant       5741
Score_medaille        0
Country               0
dtype: int64

In [42] used -1.5 MiB RAM in 0.17s (system mean cpu 18%, single max cpu 100%), peaked 1.5 MiB above final usage, current RAM usage now 225.8 MiB


In [43]:
df_complete[df_complete['Age'].isna()].groupby('Medal')['Name'].count().reset_index()

Medal  Name
0  Pas_medaille   156

In [43] used 0.1 MiB RAM in 0.11s (system mean cpu 30%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 225.9 MiB


In [44]:
#df_tokyo=pd.read_csv('data/athlete_events_tokyo_cleaned.csv')

#df_tokyo.isna().sum()

In [44] used 0.0 MiB RAM in 0.11s (system mean cpu 41%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 225.9 MiB


Enfaite il y a 156 personnes qui manquent prcq c'est les qu'on a pas les sport 

In [45]:
df_complete=df_complete.dropna(subset='Age')

In [45] used 4.4 MiB RAM in 0.13s (system mean cpu 27%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 230.2 MiB


In [46]:
df_complete.isna().sum()

Name                  0
Sex                   0
Age                   0
Height             7502
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Medal                 0
Event              9093
ID                11328
Weight            12538
Country_org           0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [46] used -1.1 MiB RAM in 0.17s (system mean cpu 24%, single max cpu 100%), peaked 1.1 MiB above final usage, current RAM usage now 229.1 MiB


## Opérations sur les poids et les tailles 

In [47]:
df_complete.isna().sum()

Name                  0
Sex                   0
Age                   0
Height             7502
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Medal                 0
Event              9093
ID                11328
Weight            12538
Country_org           0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [47] used -0.9 MiB RAM in 0.16s (system mean cpu 11%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 228.2 MiB


In [48]:
table_taille=df_complete[df_complete['Height'].isna()==False].groupby(['Sport'])['Height'].mean().apply(m.ceil).reset_index()

liste_sport=table_taille['Sport'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_taille=table_taille['Height'].tolist()
print(table_taille)
liste_sport_sex=[]
for i in range(len(liste_sport)):
    liste_sport_sex.append((liste_sport[i],liste_taille[i]))
#print(liste_sport_sex)

dictio_sport_sex=dict(zip(liste_sport,liste_taille))



                  Sport  Height
0        3x3 Basketball     188
1         Alpine Skiing     176
2               Archery     174
3   Artistic Gymnastics     162
4     Artistic Swimming     172
..                  ...     ...
62            Triathlon     174
63           Volleyball     190
64           Water Polo     186
65        Weightlifting     167
66            Wrestling     174

[67 rows x 2 columns]
In [48] used 10.0 MiB RAM in 0.14s (system mean cpu 23%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 238.3 MiB


In [49]:
df_complete['Height'] = df_complete['Height'].fillna(df_complete['Sport'].map(dictio_sport_sex))


df_complete.isna().sum()

<ipython-input-49-30982cdfd39c>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_complete['Height'] = df_complete['Height'].fillna(df_complete['Sport'].map(dictio_sport_sex))


Name                  0
Sex                   0
Age                   0
Height              159
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Medal                 0
Event              9093
ID                11328
Weight            12538
Country_org           0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [49] used 0.7 MiB RAM in 0.17s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 239.0 MiB


In [50]:
#df_complete[df_complete['Height'].isna()]

In [50] used 0.1 MiB RAM in 0.11s (system mean cpu 18%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 239.1 MiB


In [51]:
table_poids=df_complete[df_complete['Weight'].isna()==False].groupby(['Sport'])['Weight'].mean().apply(m.ceil).reset_index()

liste_sport_2=table_poids['Sport'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids=table_poids['Weight'].tolist()



dictio_sport_poids=dict(zip(liste_sport_2,liste_poids))

In [51] used 9.7 MiB RAM in 0.14s (system mean cpu 25%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 248.8 MiB


In [52]:
df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Sport'].map(dictio_sport_poids))


df_complete.info()

<class 'pandas.core.frame.DataFrame'>
Index: 96574 entries, 0 to 282626
Data columns (total 20 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Name            96574 non-null  object 
 1   Sex             96574 non-null  object 
 2   Age             96574 non-null  float64
 3   Height          96415 non-null  float64
 4   Team            96574 non-null  object 
 5   NOC             96574 non-null  object 
 6   Games           96574 non-null  object 
 7   Year            96574 non-null  int64  
 8   Season          96574 non-null  object 
 9   City            96574 non-null  object 
 10  Sport           96574 non-null  object 
 11  Medal           96574 non-null  object 
 12  Event           87481 non-null  object 
 13  ID              85246 non-null  float64
 14  Weight          94584 non-null  float64
 15  Country_org     96574 non-null  object 
 16  Population      93728 non-null  float64
 17  PIB/habitant    90843 non-null  flo

<ipython-input-52-f91662992dfc>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Sport'].map(dictio_sport_poids))


In [53]:
df_complete=df_complete[df_complete['Season']=='Summer']

df_complete

Name Sex   Age  Height       Team  NOC  \
0                    AALERUD Katrine   F  27.0   175.0     Norway  NOR   
1                        ABAD Nestor   M  28.0   165.0      Spain  ESP   
2                  ABAGNALE Giovanni   M  26.0   198.0      Italy  ITA   
3                     ABALDE Alberto   M  26.0   200.0      Spain  ESP   
4                      ABALDE Tamara   F  32.0   192.0      Spain  ESP   
...                              ...  ..   ...     ...        ...  ...   
282616      Fernando scar Zylberberg   M  23.0   168.0  Argentina  ARG   
282617      Fernando scar Zylberberg   M  27.0   168.0  Argentina  ARG   
282619  Aleksandr Viktorovich Zyuzin   M  24.0   183.0     Russia  RUS   
282620  Aleksandr Viktorovich Zyuzin   M  28.0   183.0     Russia  RUS   
282621        Olga Igorevna Zyuzkova   F  33.0   171.0    Belarus  BLR   

              Games  Year  Season            City                Sport  \
0        Tokyo 2020  2020  Summer           Tokyo         Cycling Road   
1        Tokyo 2020  2020  Summer           Tokyo  Artistic Gymnastics   
2        Tokyo 2020  2020  Summer           Tokyo               Rowing   
3        Tokyo 2020  2020  Summer           Tokyo           Basketball   
4        Tokyo 2020  2020  Summer           Tokyo           Basketball   
...             ...   ...     ...             ...                  ...   
282616  2000 Summer  2000  Summer          Sydney               Hockey   
282617  2004 Summer  2004  Summer          Athina               Hockey   
282619  2000 Summer  2000  Summer          Sydney               Rowing   
282620  2004 Summer  2004  Summer          Athina               Rowing   
282621  2016 Summer  2016  Summer  Rio de Janeiro           Basketball   

               Medal                                   Event        ID  \
0       Pas_medaille                                     NaN       NaN   
1       Pas_medaille                                     NaN       NaN   
2       Pas_medaille                                     NaN       NaN   
3       Pas_medaille                                     NaN       NaN   
4       Pas_medaille                                     NaN       NaN   
...              ...                                     ...       ...   
282616  Pas_medaille                     Hockey Men's Hockey  135565.0   
282617  Pas_medaille                     Hockey Men's Hockey  135565.0   
282619  Pas_medaille  Rowing Men's Lightweight Coxless Fours  135567.0   
282620  Pas_medaille  Rowing Men's Lightweight Coxless Fours  135567.0   
282621  Pas_medaille           Basketball Women's Basketball  135568.0   

        Weight Country_org   Population  PIB/habitant  Score_medaille  \
0          NaN       Japan    5195921.0  74400.369777               0   
1          NaN       Japan   46418269.0  25831.582305               0   
2         81.0       Japan   60802085.0  29957.804315               0   
3         88.0       Japan   46418269.0  25831.582305               0   
4         88.0       Japan   46418269.0  25831.582305               0   
...        ...         ...          ...           ...             ...   
282616    76.0   Australia   43416755.0  13431.878340               0   
282617    76.0      Greece   43416755.0  13431.878340               0   
282619    72.0   Australia  144096812.0   9092.580536               0   
282620    72.0      Greece  144096812.0   9092.580536               0   
282621    69.0      Brazil    9513000.0   5740.456495               0   

          Country  
0          Norway  
1           Spain  
2           Italy  
3           Spain  
4           Spain  
...           ...  
282616  Argentina  
282617  Argentina  
282619     Russia  
282620     Russia  
282621    Belarus  

[78790 rows x 20 columns]

In [53] used 3.9 MiB RAM in 0.14s (system mean cpu 50%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 249.8 MiB


## Moyenne par pays pour les sports qu'on ne connait pas 


In [54]:
table_pays_1=df_complete[df_complete['Weight'].isna()==False].groupby(['Country'])['Weight'].mean().apply(m.ceil).reset_index()

liste_pays_1=table_pays_1['Country'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids_1=table_pays_1['Weight'].tolist()



dictio_pays_poids_1=dict(zip(liste_pays_1,liste_poids_1))
df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Country'].map(dictio_pays_poids_1))

In [54] used 4.4 MiB RAM in 0.14s (system mean cpu 24%, single max cpu 71%), peaked 0.0 MiB above final usage, current RAM usage now 254.2 MiB


In [55]:
table_pays_2=df_complete[df_complete['Height'].isna()==False].groupby(['Country'])['Height'].mean().apply(m.ceil).reset_index()

liste_pays_2=table_pays_2['Country'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids_2=table_pays_2['Height'].tolist()



dictio_pays_poids_2=dict(zip(liste_pays_2,liste_poids_2))
df_complete['Height'] = df_complete['Height'].fillna(df_complete['Country'].map(dictio_pays_poids_2))

#df_complete.isna().sum()

In [55] used 7.2 MiB RAM in 0.14s (system mean cpu 29%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 261.4 MiB


In [56]:
#df_complete.to_csv('data/df_merge_FLO.csv', index=False)

In [56] used -17.8 MiB RAM in 0.11s (system mean cpu 99%, single max cpu 100%), peaked 17.8 MiB above final usage, current RAM usage now 243.5 MiB


## Web Scrapping FLo 

In [57]:
from bs4 import BeautifulSoup
import requests

In [57] used -120.3 MiB RAM in 0.34s (system mean cpu 39%, single max cpu 100%), peaked 91.1 MiB above final usage, current RAM usage now 123.3 MiB


In [58]:
url='https://www.topendsports.com/events/summer/hosts/list.htm'
response=requests.get(url)

In [58] used 1.1 MiB RAM in 1.23s (system mean cpu 17%, single max cpu 100%), peaked 3.3 MiB above final usage, current RAM usage now 124.4 MiB


In [59]:
url_2='https://en.wikipedia.org/wiki/List_of_Olympic_Games_host_cities'

In [59] used 0.1 MiB RAM in 0.11s (system mean cpu 20%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 124.5 MiB


In [60]:
response_2=requests.get(url_2)


In [60] used -13.3 MiB RAM in 0.24s (system mean cpu 33%, single max cpu 100%), peaked 13.3 MiB above final usage, current RAM usage now 111.1 MiB


In [61]:
soup_2=BeautifulSoup(response_2.text,'html.parser')


In [61] used -8.7 MiB RAM in 0.26s (system mean cpu 31%, single max cpu 100%), peaked 0.3 MiB above final usage, current RAM usage now 102.4 MiB


## On scrapp les pays organisateurs 

In [62]:
L_pays=[]
L_annees=[]
elements=soup_2.findAll('span',class_='datasortkey')

for element in elements:
    pays=element.find('a').text
    L_pays.append(pays)

L_pays=L_pays[42:58]




In [62] used 0.1 MiB RAM in 0.12s (system mean cpu 27%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 102.5 MiB


In [63]:
L_annees_2=[2000 + i  for i in range(0,34,2)]
L_annees_2.pop(-2)

L_annees_2

[2000,
 2002,
 2004,
 2006,
 2008,
 2010,
 2012,
 2014,
 2016,
 2018,
 2020,
 2022,
 2024,
 2026,
 2028,
 2032]

In [63] used -23.8 MiB RAM in 0.11s (system mean cpu 43%, single max cpu 100%), peaked 15.5 MiB above final usage, current RAM usage now 78.8 MiB


In [64]:
L_pays_annee=[]
for i in range(len(L_pays)):
    L_pays_annee.append((L_pays[i],L_annees_2[i]))

df_hote=pd.DataFrame(L_pays_annee,columns=['Host_country','Year'])

df_hote




Host_country  Year
0        Australia  2000
1    United States  2002
2           Greece  2004
3            Italy  2006
4            China  2008
5           Canada  2010
6   United Kingdom  2012
7           Russia  2014
8           Brazil  2016
9      South Korea  2018
10           Japan  2020
11           China  2022
12          France  2024
13           Italy  2026
14   United States  2028
15       Australia  2032

In [64] used -1.9 MiB RAM in 0.12s (system mean cpu 59%, single max cpu 100%), peaked 1.8 MiB above final usage, current RAM usage now 76.9 MiB


In [65]:
df_complete

Name Sex   Age  Height       Team  NOC  \
0                    AALERUD Katrine   F  27.0   175.0     Norway  NOR   
1                        ABAD Nestor   M  28.0   165.0      Spain  ESP   
2                  ABAGNALE Giovanni   M  26.0   198.0      Italy  ITA   
3                     ABALDE Alberto   M  26.0   200.0      Spain  ESP   
4                      ABALDE Tamara   F  32.0   192.0      Spain  ESP   
...                              ...  ..   ...     ...        ...  ...   
282616      Fernando scar Zylberberg   M  23.0   168.0  Argentina  ARG   
282617      Fernando scar Zylberberg   M  27.0   168.0  Argentina  ARG   
282619  Aleksandr Viktorovich Zyuzin   M  24.0   183.0     Russia  RUS   
282620  Aleksandr Viktorovich Zyuzin   M  28.0   183.0     Russia  RUS   
282621        Olga Igorevna Zyuzkova   F  33.0   171.0    Belarus  BLR   

              Games  Year  Season            City                Sport  \
0        Tokyo 2020  2020  Summer           Tokyo         Cycling Road   
1        Tokyo 2020  2020  Summer           Tokyo  Artistic Gymnastics   
2        Tokyo 2020  2020  Summer           Tokyo               Rowing   
3        Tokyo 2020  2020  Summer           Tokyo           Basketball   
4        Tokyo 2020  2020  Summer           Tokyo           Basketball   
...             ...   ...     ...             ...                  ...   
282616  2000 Summer  2000  Summer          Sydney               Hockey   
282617  2004 Summer  2004  Summer          Athina               Hockey   
282619  2000 Summer  2000  Summer          Sydney               Rowing   
282620  2004 Summer  2004  Summer          Athina               Rowing   
282621  2016 Summer  2016  Summer  Rio de Janeiro           Basketball   

               Medal                                   Event        ID  \
0       Pas_medaille                                     NaN       NaN   
1       Pas_medaille                                     NaN       NaN   
2       Pas_medaille                                     NaN       NaN   
3       Pas_medaille                                     NaN       NaN   
4       Pas_medaille                                     NaN       NaN   
...              ...                                     ...       ...   
282616  Pas_medaille                     Hockey Men's Hockey  135565.0   
282617  Pas_medaille                     Hockey Men's Hockey  135565.0   
282619  Pas_medaille  Rowing Men's Lightweight Coxless Fours  135567.0   
282620  Pas_medaille  Rowing Men's Lightweight Coxless Fours  135567.0   
282621  Pas_medaille           Basketball Women's Basketball  135568.0   

        Weight Country_org   Population  PIB/habitant  Score_medaille  \
0         74.0       Japan    5195921.0  74400.369777               0   
1         71.0       Japan   46418269.0  25831.582305               0   
2         81.0       Japan   60802085.0  29957.804315               0   
3         88.0       Japan   46418269.0  25831.582305               0   
4         88.0       Japan   46418269.0  25831.582305               0   
...        ...         ...          ...           ...             ...   
282616    76.0   Australia   43416755.0  13431.878340               0   
282617    76.0      Greece   43416755.0  13431.878340               0   
282619    72.0   Australia  144096812.0   9092.580536               0   
282620    72.0      Greece  144096812.0   9092.580536               0   
282621    69.0      Brazil    9513000.0   5740.456495               0   

          Country  
0          Norway  
1           Spain  
2           Italy  
3           Spain  
4           Spain  
...           ...  
282616  Argentina  
282617  Argentina  
282619     Russia  
282620     Russia  
282621    Belarus  

[78790 rows x 20 columns]

In [65] used -3.9 MiB RAM in 0.13s (system mean cpu 76%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 73.0 MiB


In [66]:
df_complete=pd.merge(df_complete,df_hote,on='Year',how='left')

df_complete

Name Sex   Age  Height       Team  NOC  \
0                   AALERUD Katrine   F  27.0   175.0     Norway  NOR   
1                       ABAD Nestor   M  28.0   165.0      Spain  ESP   
2                 ABAGNALE Giovanni   M  26.0   198.0      Italy  ITA   
3                    ABALDE Alberto   M  26.0   200.0      Spain  ESP   
4                     ABALDE Tamara   F  32.0   192.0      Spain  ESP   
...                             ...  ..   ...     ...        ...  ...   
78785      Fernando scar Zylberberg   M  23.0   168.0  Argentina  ARG   
78786      Fernando scar Zylberberg   M  27.0   168.0  Argentina  ARG   
78787  Aleksandr Viktorovich Zyuzin   M  24.0   183.0     Russia  RUS   
78788  Aleksandr Viktorovich Zyuzin   M  28.0   183.0     Russia  RUS   
78789        Olga Igorevna Zyuzkova   F  33.0   171.0    Belarus  BLR   

             Games  Year  Season            City  ...         Medal  \
0       Tokyo 2020  2020  Summer           Tokyo  ...  Pas_medaille   
1       Tokyo 2020  2020  Summer           Tokyo  ...  Pas_medaille   
2       Tokyo 2020  2020  Summer           Tokyo  ...  Pas_medaille   
3       Tokyo 2020  2020  Summer           Tokyo  ...  Pas_medaille   
4       Tokyo 2020  2020  Summer           Tokyo  ...  Pas_medaille   
...            ...   ...     ...             ...  ...           ...   
78785  2000 Summer  2000  Summer          Sydney  ...  Pas_medaille   
78786  2004 Summer  2004  Summer          Athina  ...  Pas_medaille   
78787  2000 Summer  2000  Summer          Sydney  ...  Pas_medaille   
78788  2004 Summer  2004  Summer          Athina  ...  Pas_medaille   
78789  2016 Summer  2016  Summer  Rio de Janeiro  ...  Pas_medaille   

                                        Event        ID  Weight  Country_org  \
0                                         NaN       NaN    74.0        Japan   
1                                         NaN       NaN    71.0        Japan   
2                                         NaN       NaN    81.0        Japan   
3                                         NaN       NaN    88.0        Japan   
4                                         NaN       NaN    88.0        Japan   
...                                       ...       ...     ...          ...   
78785                     Hockey Men's Hockey  135565.0    76.0    Australia   
78786                     Hockey Men's Hockey  135565.0    76.0       Greece   
78787  Rowing Men's Lightweight Coxless Fours  135567.0    72.0    Australia   
78788  Rowing Men's Lightweight Coxless Fours  135567.0    72.0       Greece   
78789           Basketball Women's Basketball  135568.0    69.0       Brazil   

        Population  PIB/habitant  Score_medaille    Country Host_country  
0        5195921.0  74400.369777               0     Norway        Japan  
1       46418269.0  25831.582305               0      Spain        Japan  
2       60802085.0  29957.804315               0      Italy        Japan  
3       46418269.0  25831.582305               0      Spain        Japan  
4       46418269.0  25831.582305               0      Spain        Japan  
...            ...           ...             ...        ...          ...  
78785   43416755.0  13431.878340               0  Argentina    Australia  
78786   43416755.0  13431.878340               0  Argentina       Greece  
78787  144096812.0   9092.580536               0     Russia    Australia  
78788  144096812.0   9092.580536               0     Russia       Greece  
78789    9513000.0   5740.456495               0    Belarus       Brazil  

[78790 rows x 21 columns]

In [66] used 79.9 MiB RAM in 0.20s (system mean cpu 59%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 153.0 MiB


In [67]:
compare_col=lambda x : 1 if x['Country']==x['Host_country'] else 0 




In [67] used -2.2 MiB RAM in 0.10s (system mean cpu 65%, single max cpu 100%), peaked 2.2 MiB above final usage, current RAM usage now 150.7 MiB


In [68]:
df_complete['Participation_own_country']=df_complete.apply(compare_col,axis=1)

df_complete[df_complete['Participation_own_country']==1]

Name Sex   Age  Height    Team  NOC        Games  Year  \
38         ABE Takatoshi   M  30.0   177.0   Japan  JPN   Tokyo 2020  2020   
79         ADACHI Kazuya   M  31.0   171.0   Japan  JPN   Tokyo 2020  2020   
80          ADACHI Seiya   M  26.0   172.0   Japan  JPN   Tokyo 2020  2020   
109          AGARIE Yuto   M  28.0   184.0   Japan  JPN   Tokyo 2020  2020   
148         AIZAWA Akira   M  24.0   177.0   Japan  JPN   Tokyo 2020  2020   
...                  ...  ..   ...     ...     ...  ...          ...   ...   
78604            Zou Kai   M  20.0   158.0   China  CHN  2008 Summer  2008   
78605            Zou Kai   M  20.0   158.0   China  CHN  2008 Summer  2008   
78606            Zou Kai   M  20.0   158.0   China  CHN  2008 Summer  2008   
78612        Zou Shiming   M  27.0   165.0   China  CHN  2008 Summer  2008   
78782  Stavroula Zygouri   F  36.0   171.0  Greece  GRE  2004 Summer  2004   

       Season     City  ...                                      Event  \
38     Summer    Tokyo  ...                                        NaN   
79     Summer    Tokyo  ...                                        NaN   
80     Summer    Tokyo  ...                                        NaN   
109    Summer    Tokyo  ...                                        NaN   
148    Summer    Tokyo  ...                                        NaN   
...       ...      ...  ...                                        ...   
78604  Summer  Beijing  ...           Gymnastics Men's Team All-Around   
78605  Summer  Beijing  ...            Gymnastics Men's Floor Exercise   
78606  Summer  Beijing  ...            Gymnastics Men's Horizontal Bar   
78612  Summer  Beijing  ...               Boxing Men's Light-Flyweight   
78782  Summer   Athina  ...  Wrestling Women's Middleweight, Freestyle   

             ID Weight                 Country_org    Population  \
38          NaN   70.0                       Japan  1.269585e+08   
79          NaN   66.0                       Japan  1.269585e+08   
80          NaN   85.0                       Japan  1.269585e+08   
109         NaN   82.0                       Japan  1.269585e+08   
148         NaN   70.0                       Japan  1.269585e+08   
...         ...    ...                         ...           ...   
78604  135269.0   55.0  People's Republic of China  1.371220e+09   
78605  135269.0   55.0  People's Republic of China  1.371220e+09   
78606  135269.0   55.0  People's Republic of China  1.371220e+09   
78612  135270.0   49.0  People's Republic of China  1.371220e+09   
78782  135560.0   63.0                      Greece  1.082373e+07   

       PIB/habitant  Score_medaille  Country  Host_country  \
38     32477.215145               0    Japan         Japan   
79     32477.215145               0    Japan         Japan   
80     32477.215145               0    Japan         Japan   
109    32477.215145               0    Japan         Japan   
148    32477.215145               0    Japan         Japan   
...             ...             ...      ...           ...   
78604   8027.683810               3    China         China   
78605   8027.683810               3    China         China   
78606   8027.683810               3    China         China   
78612   8027.683810               3    China         China   
78782  18002.230578               0   Greece        Greece   

      Participation_own_country  
38                            1  
79                            1  
80                            1  
109                           1  
148                           1  
...                         ...  
78604                         1  
78605                         1  
78606                         1  
78612                         1  
78782                         1  

[3899 rows x 22 columns]

In [68] used 10.2 MiB RAM in 0.79s (system mean cpu 63%, single max cpu 100%), peaked 8.3 MiB above final usage, current RAM usage now 160.9 MiB


In [69]:
table_sports=df_complete.groupby('Sport')['Name'].count().reset_index()

table_sports

Sport   Name
0          3x3 Basketball     56
1                 Archery   1139
2     Artistic Gymnastics    191
3       Artistic Swimming    100
4               Athletics  13711
5               Badminton   1135
6                Baseball    574
7       Baseball/Softball    234
8              Basketball   1715
9        Beach Volleyball    570
10                 Boxing   1711
11           Canoe Slalom     80
12           Canoe Sprint    243
13               Canoeing   2177
14                Cycling   3192
15  Cycling BMX Freestyle     16
16     Cycling BMX Racing     42
17  Cycling Mountain Bike     74
18           Cycling Road    189
19          Cycling Track    224
20                 Diving   1114
21             Equestrian    232
22          Equestrianism   1711
23                Fencing   1937
24               Football   2807
25                   Golf    240
26             Gymnastics   5000
27               Handball   2034
28                 Hockey   2272
29                   Judo   2330
30                 Karate     79
31      Marathon Swimming     47
32      Modern Pentathlon    391
33    Rhythmic Gymnastics    543
34                 Rowing   3282
35           Rugby Sevens    598
36                Sailing   2301
37               Shooting   3190
38          Skateboarding     79
39               Softball    358
40         Sport Climbing     36
41                Surfing     40
42               Swimming   9124
43  Synchronized Swimming    586
44           Table Tennis   1425
45              Taekwondo    728
46                 Tennis   1508
47  Trampoline Gymnastics     28
48           Trampolining    152
49              Triathlon    637
50             Volleyball   1677
51             Water Polo   1530
52          Weightlifting   1446
53              Wrestling   1955

In [69] used 0.9 MiB RAM in 0.12s (system mean cpu 53%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 161.8 MiB


## Dictionnaire group sport

On va essayer de scrapper les sports individuels et les collectifs 


In [70]:
url_sports='https://www.britannica.com/topic/list-of-sports-2038581#ref330452'

In [70] used -5.9 MiB RAM in 0.11s (system mean cpu 49%, single max cpu 100%), peaked 5.9 MiB above final usage, current RAM usage now 155.8 MiB


In [71]:
response_sp=requests.get(url_sports)

In [71] used 3.8 MiB RAM in 0.52s (system mean cpu 25%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 159.7 MiB


In [72]:
soup_sports=BeautifulSoup(response_sp.text,'html.parser')


In [72] used 74.4 MiB RAM in 0.23s (system mean cpu 12%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.0 MiB


In [73]:
sports=soup_sports.findAll('div')
L_sports_indiv=[]
for sport in sports:
    a=sport.find('a',class_='md-crosslink')
    if a is not None:
        L_sports_indiv.append(a.text)
   
    
L_sports_collec=[]

sports_co=soup_sports.findAll('section',id='ref330452')

for sport in sports_co:
    b=sport.find('a',class_='md-crosslink')
    print(b)
   
        
#sports

<a class="md-crosslink" data-show-preview="true" href="https://www.britannica.com/sports/association-croquet">association croquet</a>
In [73] used 0.0 MiB RAM in 0.13s (system mean cpu 59%, single max cpu 67%), peaked 0.0 MiB above final usage, current RAM usage now 234.1 MiB


## Pays passé communiste 

In [74]:
url_soviet='https://en.wikipedia.org/wiki/Post-Soviet_states'

In [74] used 0.1 MiB RAM in 0.11s (system mean cpu 49%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.1 MiB


In [75]:
response_soviet=requests.get(url_soviet)

In [75] used 0.3 MiB RAM in 0.19s (system mean cpu 28%, single max cpu 62%), peaked 0.0 MiB above final usage, current RAM usage now 234.5 MiB


In [76]:
soup_soviet=BeautifulSoup(response_soviet.text,'html.parser')

In [76] used -19.7 MiB RAM in 0.35s (system mean cpu 32%, single max cpu 100%), peaked 19.8 MiB above final usage, current RAM usage now 214.8 MiB


In [77]:
L_pays_soviet=[]

pays_soviet=soup_soviet.findAll('b')
#print(pays_soviet)
for pays in pays_soviet:
    b=pays.find('a')
    if b is not None:
        L_pays_soviet.append(b.text)
L_pays_soviet=L_pays_soviet[:15]

L_pays_soviet

['Kazakhstan',
 'Kyrgyzstan',
 'Tajikistan',
 'Turkmenistan',
 'Uzbekistan',
 'Belarus',
 'Moldova',
 'Russia',
 'Ukraine',
 'Estonia',
 'Latvia',
 'Lithuania',
 'Armenia',
 'Azerbaijan',
 'Georgia']

In [77] used 0.1 MiB RAM in 0.12s (system mean cpu 14%, single max cpu 71%), peaked 0.0 MiB above final usage, current RAM usage now 214.9 MiB


In [78]:
complete_urss_regime=lambda x : 1 if x in L_pays_soviet else 0

In [78] used 0.0 MiB RAM in 0.10s (system mean cpu 28%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 214.9 MiB


In [79]:
df_complete['Soviet_past']=df_complete['Country'].apply(complete_urss_regime)



In [79] used 3.8 MiB RAM in 0.16s (system mean cpu 47%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 218.6 MiB


In [80]:
df_complete.isna().sum()

Name                             0
Sex                              0
Age                              0
Height                           0
Team                             0
NOC                              0
Games                            0
Year                             0
Season                           0
City                             0
Sport                            0
Medal                            0
Event                         9093
ID                           11328
Weight                           0
Country_org                      0
Population                    2609
PIB/habitant                  5315
Score_medaille                   0
Country                          0
Host_country                     0
Participation_own_country        0
Soviet_past                      0
dtype: int64

In [80] used 15.0 MiB RAM in 0.16s (system mean cpu 25%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 233.7 MiB


In [81]:
df_pop_na=df_complete[df_complete['Population'].isna()]

#df_pop_na.groupby('Country')['Name'].count().reset_index()

In [81] used 0.3 MiB RAM in 0.11s (system mean cpu 26%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 233.9 MiB


## Scrap Population et GPA 

In [82]:
import re

In [82] used 0.0 MiB RAM in 0.10s (system mean cpu 24%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.0 MiB


In [83]:
url_pop_gpa='https://www.worldometers.info/gdp/gdp-by-country/'

In [83] used 0.0 MiB RAM in 0.11s (system mean cpu 30%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.0 MiB


In [84]:
response_pop_gpa=requests.get(url_pop_gpa)


In [84] used 0.1 MiB RAM in 0.88s (system mean cpu 26%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.1 MiB


In [85]:
soup_pop_gpa=BeautifulSoup(response_pop_gpa.text,'html.parser')


In [85] used 0.0 MiB RAM in 0.18s (system mean cpu 33%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.1 MiB


In [86]:
L_pays=[]

all_countries=soup_pop_gpa.findAll('td')

for country in all_countries:
    a=country.find('a')
    if a is not None:
        L_pays.append(a.text)
len(L_pays)

177

In [86] used 0.0 MiB RAM in 0.12s (system mean cpu 18%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.1 MiB


In [87]:
#b=re.findall(r'\d+\,?\d*|\d+\.\d+', info.text)

In [87] used 0.3 MiB RAM in 0.11s (system mean cpu 16%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.4 MiB


In [88]:
all_info=soup_pop_gpa.findAll('td',attrs={'style':'font-weight: bold; text-align:right;'})
#print(all_info)
numbers=[]
for i,info in enumerate(all_info):
    b=re.findall(r'\d+\b', info.text)
    b=''.join(b)
    numbers.append(b)
   
 
    #b=[number.replace(',','.') for number in b]
#print(len(numbers))
population=[]
gpd_per_habitant=[]
for i in range(3,len(numbers),6):
    population.append(int(numbers[i]))
for i in range(4,len(numbers),6):
    gpd_per_habitant.append(int(numbers[i]))
    
print(len(population))
print(len(gpd_per_habitant))

177
177
In [88] used 0.0 MiB RAM in 0.13s (system mean cpu 27%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.4 MiB


In [89]:
df_pop_gpd=pd.DataFrame({'Country':L_pays,'Population':population,'PIB/habitant':gpd_per_habitant})

df_pop_gpd['Country']=df_pop_gpd['Country'].replace('Trinidad and Tobago','Trinidad')
df_pop_gpd['Country']=df_pop_gpd['Country'].replace('State of Palestine','Palestine, Occupied Territories')
df_pop_gpd['Country']=df_pop_gpd['Country'].replace('Singapore','Singapour')

#new_row=pd.DataFrame({'Country':'Taiwan','Population':23923276,'PIB/habitant':44460})
#df_pop_gpd=df_pop_gpd.append(new_row,ignore_index=True)

df_pop_gpd.loc[1]=['Taiwan',23923276,44460]
df_pop_gpd.loc[2]=['Cook Islands',17044,13558]
df_pop_gpd.loc[3]=['Eritrea',3684000,1067]
df_pop_gpd.loc[4]=['Kosovo',1760000,5340]
df_pop_gpd.loc[5]=['Liban',5425542,4136]
df_pop_gpd.loc[6]=['Netherlands Antilles*',27148,11400]
df_pop_gpd.loc[7]=['South Sudan',11000000,467]








df_pop_gpd[df_pop_gpd['Country']=='Iran']


Country  Population  PIB/habitant
39    Iran    88550570          4388

In [89] used 0.1 MiB RAM in 0.11s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.6 MiB


In [90]:
df_pop_gpd=pd.concat([df_pop_gpd,pd.DataFrame([['American Samoa*',43914,19673],
                                              ['Aruba*',106445,333000],
                                              ['Bermuda*',63532,118774],
                                               ['British Virgin Islands',31707,34246],
                                               ['Cayman Islands*',69310,99624],
                                               ['Cuba',11179471,56495],
                                               ['Guam',173825,35904],
                                               ['Korea, North',26070000,33719],
                                               ['Liechtenstein',39039,197504],
                                               ['Monaco',39050,240862],
                                               ['Puerto Rico*',3260314,35208],
                                               ['San Marino',33642,53000],
                                               ['Syria',22130000,420],
                                               ['Venezuela',29254044,2624],
                                               ['Virgin Islands*',105413,41976]
                                              ],
                                              columns=['Country','Population','PIB/habitant'])],ignore_index=True)

In [90] used 0.1 MiB RAM in 0.11s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.7 MiB


In [91]:
df_pop_gpd

Country  Population  PIB/habitant
0      United States   338289857         75269
1             Taiwan    23923276         44460
2       Cook Islands       17044         13558
3            Eritrea     3684000          1067
4             Kosovo     1760000          5340
..               ...         ...           ...
187     Puerto Rico*     3260314         35208
188       San Marino       33642         53000
189            Syria    22130000           420
190        Venezuela    29254044          2624
191  Virgin Islands*      105413         41976

[192 rows x 3 columns]

In [91] used -0.0 MiB RAM in 0.12s (system mean cpu 11%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.7 MiB


## Dictionnaire pour remplir les valeurs NaN de Pop et PIB sans changer les valeurs de pop et PIB per habitant précédentes

In [92]:
dictio_pop_gpd=df_pop_gpd.to_dict()

dictio_pop_gpd

{'Country': {0: 'United States',
  1: 'Taiwan',
  2: 'Cook Islands',
  3: 'Eritrea',
  4: 'Kosovo',
  5: 'Liban',
  6: 'Netherlands Antilles*',
  7: 'South Sudan',
  8: 'Canada',
  9: 'Italy',
  10: 'Brazil',
  11: 'Australia',
  12: 'South Korea',
  13: 'Mexico',
  14: 'Spain',
  15: 'Indonesia',
  16: 'Saudi Arabia',
  17: 'Netherlands',
  18: 'Turkey',
  19: 'Switzerland',
  20: 'Poland',
  21: 'Argentina',
  22: 'Sweden',
  23: 'Norway',
  24: 'Belgium',
  25: 'Ireland',
  26: 'Israel',
  27: 'United Arab Emirates',
  28: 'Thailand',
  29: 'Nigeria',
  30: 'Egypt',
  31: 'Austria',
  32: 'Singapour',
  33: 'Bangladesh',
  34: 'Vietnam',
  35: 'Malaysia',
  36: 'South Africa',
  37: 'Philippines',
  38: 'Denmark',
  39: 'Iran',
  40: 'Pakistan',
  41: 'Hong Kong',
  42: 'Colombia',
  43: 'Romania',
  44: 'Chile',
  45: 'Czech Republic (Czechia)',
  46: 'Finland',
  47: 'Iraq',
  48: 'Portugal',
  49: 'New Zealand',
  50: 'Peru',
  51: 'Qatar',
  52: 'Kazakhstan',
  53: 'Greece',
  5

In [92] used 0.2 MiB RAM in 0.12s (system mean cpu 21%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.8 MiB


In [93]:
df_test=df_complete.copy()



In [93] used 29.8 MiB RAM in 0.14s (system mean cpu 11%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 264.6 MiB


In [94]:
df_test.head(20)

Name Sex   Age  Height                      Team  NOC  \
0           AALERUD Katrine   F  27.0   175.0                    Norway  NOR   
1               ABAD Nestor   M  28.0   165.0                     Spain  ESP   
2         ABAGNALE Giovanni   M  26.0   198.0                     Italy  ITA   
3            ABALDE Alberto   M  26.0   200.0                     Spain  ESP   
4             ABALDE Tamara   F  32.0   192.0                     Spain  ESP   
5              ABAROA Cesar   M  25.0   179.0                     Chile  CHI   
6             ABASS Abobakr   M  23.0   181.0                     Sudan  SUD   
7        ABBAGNALE Vincenzo   M  28.0   187.0                     Italy  ITA   
8          ABBASALI Hamideh   F  31.0   181.0  Islamic Republic of Iran  IRI   
9             ABBASOV Islam   M  25.0   174.0                Azerbaijan  AZE   
10             ABBINGH Lois   F  29.0   177.0               Netherlands  NED   
11              ABBOT Emily   F  24.0   160.0                 Australia  AUS   
12  ABDALLA Abubaker Haydar   M  25.0   177.0                     Qatar  QAT   
13      ABDALRASOOL Mohamed   M  28.0   174.0                     Sudan  SUD   
14        ABDEL LATIF Radwa   F  32.0   173.0                     Egypt  EGY   
15         ABDEL RAZEK Samy   M  41.0   173.0                     Egypt  EGY   
16         ABDELAAL Mohamed   M  31.0   174.0                     Egypt  EGY   
17        ABDELAZIZ Abdalla   M  22.0   178.0                     Egypt  EGY   
18          ABDELAZIZ Farah   F  29.0   172.0                     Egypt  EGY   
19     ABDELMAWGOUD Mohamed   M  27.0   174.0                     Egypt  EGY   

         Games  Year  Season   City  ...  ID Weight Country_org  Population  \
0   Tokyo 2020  2020  Summer  Tokyo  ... NaN   74.0       Japan   5195921.0   
1   Tokyo 2020  2020  Summer  Tokyo  ... NaN   71.0       Japan  46418269.0   
2   Tokyo 2020  2020  Summer  Tokyo  ... NaN   81.0       Japan  60802085.0   
3   Tokyo 2020  2020  Summer  Tokyo  ... NaN   88.0       Japan  46418269.0   
4   Tokyo 2020  2020  Summer  Tokyo  ... NaN   88.0       Japan  46418269.0   
5   Tokyo 2020  2020  Summer  Tokyo  ... NaN   81.0       Japan  17948141.0   
6   Tokyo 2020  2020  Summer  Tokyo  ... NaN   73.0       Japan  40234882.0   
7   Tokyo 2020  2020  Summer  Tokyo  ... NaN   81.0       Japan  60802085.0   
8   Tokyo 2020  2020  Summer  Tokyo  ... NaN   81.0       Japan  79109272.0   
9   Tokyo 2020  2020  Summer  Tokyo  ... NaN   77.0       Japan   9651349.0   
10  Tokyo 2020  2020  Summer  Tokyo  ... NaN   82.0       Japan  16936520.0   
11  Tokyo 2020  2020  Summer  Tokyo  ... NaN   50.0       Japan  23781169.0   
12  Tokyo 2020  2020  Summer  Tokyo  ... NaN   70.0       Japan   2235355.0   
13  Tokyo 2020  2020  Summer  Tokyo  ... NaN   79.0       Japan  40234882.0   
14  Tokyo 2020  2020  Summer  Tokyo  ... NaN   74.0       Japan  91508084.0   
15  Tokyo 2020  2020  Summer  Tokyo  ... NaN   74.0       Japan  91508084.0   
16  Tokyo 2020  2020  Summer  Tokyo  ... NaN   79.0       Japan  91508084.0   
17  Tokyo 2020  2020  Summer  Tokyo  ... NaN   75.0       Japan  91508084.0   
18  Tokyo 2020  2020  Summer  Tokyo  ... NaN   66.0       Japan  91508084.0   
19  Tokyo 2020  2020  Summer  Tokyo  ... NaN   79.0       Japan  91508084.0   

    PIB/habitant Score_medaille      Country  Host_country  \
0   74400.369777              0       Norway         Japan   
1   25831.582305              0        Spain         Japan   
2   29957.804315              0        Italy         Japan   
3   25831.582305              0        Spain         Japan   
4   25831.582305              0        Spain         Japan   
5   13416.230039              0        Chile         Japan   
6    2414.723601              0        Sudan         Japan   
7   29957.804315              0        Italy         Japan   
8            NaN              0         Iran         Japan   
9    5496.344640              0   Azerbaijan         Japan   


In [94] used 0.1 MiB RAM in 0.14s (system mean cpu 19%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 264.7 MiB


## Nouvelle dinguerie 

In [95]:
df_test.isna().sum()

Name                             0
Sex                              0
Age                              0
Height                           0
Team                             0
NOC                              0
Games                            0
Year                             0
Season                           0
City                             0
Sport                            0
Medal                            0
Event                         9093
ID                           11328
Weight                           0
Country_org                      0
Population                    2609
PIB/habitant                  5315
Score_medaille                   0
Country                          0
Host_country                     0
Participation_own_country        0
Soviet_past                      0
dtype: int64

In [95] used 0.3 MiB RAM in 0.16s (system mean cpu 10%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 265.0 MiB


In [96]:
df_test_na_pop=df_test[df_test['Population'].isna()]

df_test_na_pop.isna().sum()

Name                            0
Sex                             0
Age                             0
Height                          0
Team                            0
NOC                             0
Games                           0
Year                            0
Season                          0
City                            0
Sport                           0
Medal                           0
Event                         329
ID                            391
Weight                          0
Country_org                     0
Population                   2609
PIB/habitant                 2609
Score_medaille                  0
Country                         0
Host_country                    0
Participation_own_country       0
Soviet_past                     0
dtype: int64

In [96] used 0.2 MiB RAM in 0.12s (system mean cpu 12%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 265.2 MiB


In [97]:
df_test_na_pop=pd.merge(df_test_na_pop,df_pop_gpd,on='Country',how='left')

df_test_na_pop.isna().sum()

Name                            0
Sex                             0
Age                             0
Height                          0
Team                            0
NOC                             0
Games                           0
Year                            0
Season                          0
City                            0
Sport                           0
Medal                           0
Event                         329
ID                            391
Weight                          0
Country_org                     0
Population_x                 2609
PIB/habitant_x               2609
Score_medaille                  0
Country                         0
Host_country                    0
Participation_own_country       0
Soviet_past                     0
Population_y                   17
PIB/habitant_y                 17
dtype: int64

In [97] used 0.2 MiB RAM in 0.12s (system mean cpu 9%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 265.4 MiB


In [98]:
df_test_na_pop=df_test_na_pop.drop(['Population_x'],axis=1)
df_test_na_pop=df_test_na_pop.drop(['PIB/habitant_x'],axis=1)


In [98] used 0.0 MiB RAM in 0.11s (system mean cpu 6%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 265.4 MiB


In [99]:
df_test_na_pop=df_test_na_pop.rename(columns={'Population_y':'Population'})
df_test_na_pop=df_test_na_pop.rename(columns={'PIB/habitant_y':'PIB/habitant'})

In [99] used 0.0 MiB RAM in 0.11s (system mean cpu 21%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 265.5 MiB


In [100]:
df_test_na_pop=df_test_na_pop.dropna(subset='Population')

In [100] used -3.1 MiB RAM in 0.11s (system mean cpu 22%, single max cpu 100%), peaked 3.1 MiB above final usage, current RAM usage now 262.4 MiB


In [101]:
#df_test_na_pop.isna().sum()

In [101] used 0.0 MiB RAM in 0.11s (system mean cpu 24%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 262.4 MiB


In [102]:
df_test=df_test.dropna(subset='Population')

df_test.isna().info()

<class 'pandas.core.frame.DataFrame'>
Index: 76181 entries, 0 to 78789
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype
---  ------                     --------------  -----
 0   Name                       76181 non-null  bool 
 1   Sex                        76181 non-null  bool 
 2   Age                        76181 non-null  bool 
 3   Height                     76181 non-null  bool 
 4   Team                       76181 non-null  bool 
 5   NOC                        76181 non-null  bool 
 6   Games                      76181 non-null  bool 
 7   Year                       76181 non-null  bool 
 8   Season                     76181 non-null  bool 
 9   City                       76181 non-null  bool 
 10  Sport                      76181 non-null  bool 
 11  Medal                      76181 non-null  bool 
 12  Event                      76181 non-null  bool 
 13  ID                         76181 non-null  bool 
 14  Weight                     

In [103]:
df_test=pd.concat([df_test,df_test_na_pop],ignore_index=True)



In [103] used 13.5 MiB RAM in 0.13s (system mean cpu 25%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 254.0 MiB


In [104]:
df_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78773 entries, 0 to 78772
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Name                       78773 non-null  object 
 1   Sex                        78773 non-null  object 
 2   Age                        78773 non-null  float64
 3   Height                     78773 non-null  float64
 4   Team                       78773 non-null  object 
 5   NOC                        78773 non-null  object 
 6   Games                      78773 non-null  object 
 7   Year                       78773 non-null  int64  
 8   Season                     78773 non-null  object 
 9   City                       78773 non-null  object 
 10  Sport                      78773 non-null  object 
 11  Medal                      78773 non-null  object 
 12  Event                      69680 non-null  object 
 13  ID                         67445 non-null  flo

## On s'occupe du PIB 

In [105]:
df_test_na_pib=df_test[df_test['PIB/habitant'].isna()]

In [105] used -0.1 MiB RAM in 0.11s (system mean cpu 41%, single max cpu 100%), peaked 0.1 MiB above final usage, current RAM usage now 249.7 MiB


In [106]:
df_test=df_test.dropna(subset='PIB/habitant')

In [106] used 4.1 MiB RAM in 0.13s (system mean cpu 67%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 253.8 MiB


In [107]:
df_test_na_pib=pd.merge(df_test_na_pib,df_pop_gpd,on='Country',how='left')

df_test_na_pib.isna().sum()

Name                            0
Sex                             0
Age                             0
Height                          0
Team                            0
NOC                             0
Games                           0
Year                            0
Season                          0
City                            0
Sport                           0
Medal                           0
Event                         246
ID                            280
Weight                          0
Country_org                     0
Population_x                    0
PIB/habitant_x               2706
Score_medaille                  0
Country                         0
Host_country                    0
Participation_own_country       0
Soviet_past                     0
Population_y                    0
PIB/habitant_y                  0
dtype: int64

In [107] used -4.5 MiB RAM in 0.12s (system mean cpu 47%, single max cpu 100%), peaked 4.0 MiB above final usage, current RAM usage now 249.3 MiB


In [108]:
df_test_na_pib=df_test_na_pib.drop(['Population_x'],axis=1)
df_test_na_pib=df_test_na_pib.drop(['PIB/habitant_x'],axis=1)

In [108] used -1.9 MiB RAM in 0.11s (system mean cpu 45%, single max cpu 100%), peaked 1.5 MiB above final usage, current RAM usage now 247.4 MiB


In [109]:
df_test_na_pib=df_test_na_pib.rename(columns={'Population_y':'Population'})
df_test_na_pib=df_test_na_pib.rename(columns={'PIB/habitant_y':'PIB/habitant'})



In [109] used -1.9 MiB RAM in 0.11s (system mean cpu 42%, single max cpu 100%), peaked 1.9 MiB above final usage, current RAM usage now 245.5 MiB


In [110]:
df_test_na_pib.isna().sum()

Name                           0
Sex                            0
Age                            0
Height                         0
Team                           0
NOC                            0
Games                          0
Year                           0
Season                         0
City                           0
Sport                          0
Medal                          0
Event                        246
ID                           280
Weight                         0
Country_org                    0
Score_medaille                 0
Country                        0
Host_country                   0
Participation_own_country      0
Soviet_past                    0
Population                     0
PIB/habitant                   0
dtype: int64

In [110] used -5.4 MiB RAM in 0.11s (system mean cpu 38%, single max cpu 100%), peaked 5.4 MiB above final usage, current RAM usage now 240.1 MiB


In [111]:
df_test_na_pib.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2706 entries, 0 to 2705
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Name                       2706 non-null   object 
 1   Sex                        2706 non-null   object 
 2   Age                        2706 non-null   float64
 3   Height                     2706 non-null   float64
 4   Team                       2706 non-null   object 
 5   NOC                        2706 non-null   object 
 6   Games                      2706 non-null   object 
 7   Year                       2706 non-null   int64  
 8   Season                     2706 non-null   object 
 9   City                       2706 non-null   object 
 10  Sport                      2706 non-null   object 
 11  Medal                      2706 non-null   object 
 12  Event                      2460 non-null   object 
 13  ID                         2426 non-null   float

In [112]:
df_test=pd.concat([df_test,df_test_na_pib],ignore_index=True)

In [112] used 6.8 MiB RAM in 0.12s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 213.5 MiB


In [113]:
df_test.isna().sum()

Name                             0
Sex                              0
Age                              0
Height                           0
Team                             0
NOC                              0
Games                            0
Year                             0
Season                           0
City                             0
Sport                            0
Medal                            0
Event                         9093
ID                           11328
Weight                           0
Country_org                      0
Population                       0
PIB/habitant                     0
Score_medaille                   0
Country                          0
Host_country                     0
Participation_own_country        0
Soviet_past                      0
dtype: int64

In [113] used 1.0 MiB RAM in 0.16s (system mean cpu 22%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 214.5 MiB


## Mise au format csv dataframe

In [114]:
#df_test.to_csv('data/data_FLO_without_NaN.csv',index=False)

In [114] used -22.6 MiB RAM in 1.15s (system mean cpu 23%, single max cpu 100%), peaked 24.6 MiB above final usage, current RAM usage now 191.9 MiB
